config_test.go

```diff
@@ -1,57 +1,165 @@
 package main
 
 import (
-	"path"
+	"bytes"
+	"log"
 	"testing"
+
+	. "github.com/smartystreets/goconvey/convey"
 )
 
 func Test_Load(t *testing.T) {
-	tt := []struct {
-		Path string
-		Err  error
-	}{
-		{
-			"valid_config.json",
-			nil,
-		}, {
-			"valid_config_extras.json",
-			nil,
-		}, {
-			"invalid_config_author.json",
-			ErrAuthorRequired,
-		}, {
-			"invalid_config_email.json",
-			ErrEmailRequired,
-		},
-	}
-
-	for _, tc := range tt {
-		path := path.Join("testdata", tc.Path)
-		_, err := LoadConfigFile(path)
-		if err != tc.Err {
-			t.Fatalf("Load(%v) returned an error (%v)", path, err)
+	Convey("When loading a valid config file", t, func() {
+		fs = mockFS{
```

Comment:
The line you are referring to is 20 in `messages_test.go` I believe and it's assigning to `rc`, not declaring. `rc` is a global so anything can use it. You'll also notice that it's `rc = mockClock{}` vs `rc := mockClock{}`. In that case it's the `New` function since `New` automatically sets the date time inside of it. Perhaps this is all just overkill. I was trying to figure out how to test our stuff and since we use so many concrete implementations, it because difficult / impossible to test. We can close this and carry on.